src/coreclr/jit/lowerarmarch.cpp

```diff
@@ -3523,50 +3500,25 @@ GenTree* Lowering::LowerHWIntrinsicCndSel(GenTreeHWIntrinsic* cndSelNode)
 {
     assert(cndSelNode->OperIsHWIntrinsic(NI_Sve_ConditionalSelect));
 
-    GenTree* op1 = cndSelNode->Op(1);
-    GenTree* op2 = cndSelNode->Op(2);
-    GenTree* op3 = cndSelNode->Op(3);
+    GenTree* op1         = cndSelNode->Op(1);
+    GenTree* op2         = cndSelNode->Op(2);
+    GenTree* op3         = cndSelNode->Op(3);
+    GenTree* lowerCndSel = cndSelNode;
 
     if (op1->OperIsHWIntrinsic())
     {
         GenTreeHWIntrinsic* predNode = op1->AsHWIntrinsic();
-        if (predNode->IsAllFalse())
-        {
-            // Transform:
-            //
-            // CndSel(AllFalse, op2, op3) to
-            // op3
-            op1->SetUnusedValue();
-            op2->SetUnusedValue();
-            cndSelNode->SetUnusedValue();
+        // op1 should never be Zero, because we optimized that in HIR already
+        // op1 should never be a SIMD node, because op1 takes TYP_MASK
+        assert(!predNode->IsMaskZero() && varTypeIsMask(predNode));
 
-            return op3;
-        }
-
-        if (predNode->IsAllTrue())
+        if (predNode->IsMaskAllBitsSet())
         {
             if (op2->isEmbeddedMaskingCompatibleHWIntrinsic())
```

Comment:
I'm not sure this path can be hit today

Operands are always iterated prior to the uses, so when we encounter an `isEmbeddedMaskingCompatibleHWIntrinsic` scenario, we're going to lower that to `CndSel(true, M(op), zero)` and call `ContainCheckHWIntrinsic` on it, there's no need to lower it since there's no other transforms that could happen.

We should only need to care about when we encounter a `CndSel` that contains another `CndSel(true, ..., ...)` as its op2